cirq/work/sampler.py

```diff
@@ -133,6 +134,78 @@ def sample(
 
         return pd.concat(results)
 
+    def sample_expectation_values(
+        self,
+        program: 'cirq.Circuit',
+        measurement_to_observables: Dict[
+            str, Union['cirq.PauliSumLike', List['cirq.PauliSumLike']]
+        ],
+        *,
+        num_samples: int,
+        params: 'cirq.Sweepable' = None,
+    ) -> List[List[float]]:
+        """Calculates estimated expectation values from samples of a circuit.
+
+        This method can be run on any device or simulator that supports circuit
+        sampling. Compare with `simulate_expectation_values` in simulator.py,
+        which is limited to simulators but provides exact results.
+
+        Args:
+            program: The circuit to simulate.
+            measurement_to_observables: A map of measurement keys to the
```

Comment:
so `program` has measure instructions? I think this is the wrong way to do this API. We either have to have "assume measurements happen at the end" or finish the "generalized measurement operations" roadmap item so you can just add cirq.Measure(qubits, basis=X) to the circuit